source/engine/graphics/api/vulkan/shaders/EnvironmentDataShaderResource.cpp

```diff
@@ -20,7 +20,7 @@ namespace nc::graphics
     {
         auto initialEnvironmentData = EnvironmentData{};
         initialEnvironmentData.cameraWorldPosition = Vector3{-0.0f, 4.0f, -6.4f};
-        initialEnvironmentData.skyboxTextureIndex = -1;
+        initialEnvironmentData.skyboxTextureIndex = 0; // ?
```

Comment:
Shoot - we can't actually do this here.

In Pbr.frag, we decision on whether or not to run a block of calculations on whether or not this is >-1.

```    if (environmentData.skyboxCubemapIndex > -1)```
```    {```
 ```       // Environment reflection```
 ```       vec3 I = normalize(inFragPosition - environmentData.cameraWorldPosition);```
 ```       vec3 surfaceNormal = normalize((inNormal + N)/2);```
 ```       vec3 reflected = reflect(I, surfaceNormal);```
 ```       vec3 environmentReflectionColor = SkyboxColor(environmentData.skyboxCubemapIndex, reflected);```

``` result += (F0 * (1-roughnessColor.r) * environmentReflectionColor) / 2;```
```    }```

I suggest changing to ```UINT32_MAX```

Reply:
Ok, added an issue and changed to unsigned max